sycl/include/CL/sycl/handler.hpp

```diff
@@ -865,8 +852,22 @@ class __SYCL_EXPORT handler {
   detail::enable_if_t<Reduction::accessor_mode == access::mode::read_write &&
                       Reduction::has_fast_atomics>
   parallel_for(nd_range<Dims> Range, Reduction &Redu, KernelType KernelFunc) {
-    intel::detail::reduCGFunc<KernelName>(*this, KernelFunc, Range, Redu,
-                                          Redu.MAcc);
+    if (Reduction::is_usm)
+      Redu.associateWithHandler(*this);
+    auto QueueCopy = MQueue;
```

Comment:
Please, do not use auto in such cases.

Reply:
I will fix this with 'shared_ptr_class<detail::queue_impl> QueueCopy = MQueue;' even though I do not agree with you regarding 'auto'.
The original statement was 'auto MQueueCopy = MQueue', which can be read as 'Whatever MQueue is, make a copy of it.'

One of key good features of 'auto' is "If you change the type of 'MQueue' in one place, you would not have to also update all other places where it is used, i.e. lines like the line 857"

Reply:
Fixed in additional commit.